google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,502 @@
+
+    blob_list = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+    pb = documentai.Document.pb()
+    for blob in blob_list:
+        name = blob.name.split("/")[-1]
+        if blob.name.endswith(".dp.bp"):
+            blob_as_bytes = blob.download_as_bytes()
+            if re.search(r"^doc.dp.bp", name):
+                text = pb.FromString(blob_as_bytes).text
+            elif re.search(r"^pages_*.*", name):
+                base_doc.append(blob_as_bytes)
+            elif re.search(r"^rev_*.*", name):
+                revisions_doc.append(blob_as_bytes)
+
+    return text, base_doc, revisions_doc
+
+
+def _get_base_docproto(gcs_bucket_name, gcs_prefix) -> Tuple[list, list]:
+    r"""
+    Returns base document.proto and list of revision shards.
+
+    Args:
+        gcs_bucket_name (str):
+            Required. The gcs bucket.
+
+            Format: Given `gs://{bucket_name}/{optional_folder}/{operation_id}/` where `gcs_bucket_name={bucket_name}`.
+        gcs_prefix (str):
+            Required. The prefix to the location of the target folder.
+
+            Format: Given `gs://{bucket_name}/{optional_folder}/{target_folder}` where `gcs_prefix={optional_folder}/{target_folder}`.
+    Returns:
+        Tuple[list, list]:
+            base document.proto, revision shards.
+    """
+    base_shards = []
+    revision_shards = []
+
+    text, base_bytes, revision_bytes = _get_base_and_revision_bytes(
+        gcs_bucket_name, gcs_prefix
+    )
+    page_pb = documentai.Document.Page.pb()
+    pb = documentai.Document.pb()
+    for byte in base_bytes:
+        base_shards.append(
+            documentai.Document(pages=[page_pb.FromString(byte)], text=text)
+        )
+
+    for byte in revision_bytes:
+        revision_shards.append(pb.FromString(byte))
+
+    return base_shards, revision_shards
+
+
+def _get_revised_entites(
+    entities: List[documentai.Document.Entity],
+) -> Tuple[list, list]:
+    r"""Modifies the entities using the providence field in the entities.
+
+    Args:
+        entities (List[Entity]):
+            A list of documentai.Document.Entity.
+
+    Returns:
+        Tuple[list,list]:
+            modified entities, Entity history.
+    """
+    entities_array = []
+    history = []
+    for e in entities:
+        if (
+            e.provenance.type_ == _OP_TYPE.ADD
+            or e.provenance.type_ == _OP_TYPE.OPERATION_TYPE_UNSPECIFIED
+        ):
+            entities_array.append(Entity(documentai_entity=e))
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                }
+            )
+        elif e.provenance.type_ == _OP_TYPE.REMOVE:
+            entity = entities_array.pop(int(e.id) - 1)
```

Comment:
This kind of operation is pretty tricky, popping from `entities_array`, which is added to or modified by the other branches.  Can we start by expanding the docstring to clarify what this function intends to accomplish?